### DISCUSSION TITLE
Should we use unchecked arithmetic in the sender amount calculations?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#882] [Ideas]
@smol-ninja's question here remained unanswered:

https://github.com/sablier-labs/v2-core/pull/813#discussion_r1486648845

Some historical context: we used to use `unchecked`, but then Cantina (Zach and Rusty) recommended removing it during the 1st audit of the code base:

https://github.com/cantinasec/review-sablier/issues/3

But IIRC, we didn't have this if block at the time when Cantina's feedback was given:

https://github.com/sablier-labs/v2-core/blob/76e29fee33befaeb54a82473c28cf182e6f8855d/src/abstracts/SablierV2Lockup.sol#L559-L561

As Shub pointed out, it should be completely safe to use `unchecked` arithmetic for calculating the sender amount.

WDYT @andreivladbrg?

**Comment by smol-ninja** [comment]
Bumping this up @andreivladbrg since Cantina audit is starting from Apr 15.

**Comment by PaulRBerg** [comment]
@andreivladbrg can we please get an answer from you here

**Comment by andreivladbrg** [comment]
Completely sorry. I have my notifications active and still got no mail for this...

If it happens in the future to not answer again, please ping me via **Slack DM**

> But IIRC, we didn't have this if block at the time when Cantina's feedback was given:

yes, we didn't

> it should be completely safe to use unchecked arithmetic for calculating the sender amount

agree, we can use safely use `unchecked`

**Reply by PaulRBerg** [reply]
No worries. @smol-ninja can you put together a PR for this, please.

**Reply by smol-ninja** [reply]
Sure!

**Comment by smol-ninja** [comment]
Resolved in:

- https://github.com/sablier-labs/v2-core/pull/892
